Employee Management/EmployeeSearchAndSort.js

```diff
@@ -0,0 +1,247 @@
+  
+  // Validate the orderBy and direction parameters
+  let validOrderBy = ['name', 'email', 'age', 'dob'];
+  if (!validOrderBy.includes(orderBy)) {
+    throw new Error('Invalid orderBy.');
+  }
+
+  let validDirection = ['asc', 'desc'];
+  if (!validDirection.includes(direction)) {
+    throw new Error('Invalid direction.');
+  }
+
+  const employees= loadEmployeesFromFile();
+  let results = employees.filter((employee) =>
+    JSON.stringify(employee).toLowerCase().includes(query.toLowerCase())
+  );
+  if (orderBy) {
+    results = results.sort((a, b) => {
+      const aProp = a[orderBy];
+      const bProp = b[orderBy];
+      if (typeof aProp === 'string') {
+        return direction === 'asc'
+          ? aProp.localeCompare(bProp)
+          : bProp.localeCompare(aProp);
+      } else {
+        return direction === 'asc' ? aProp - bProp : bProp - aProp;
+      }
+    });
+  }
+  return results;
+}
+
+// Display all employees
+function loadEmployeesFromFile(){
+  let employees = [];
+
+  try{
+  const data = fs.readFileSync(fileName, 'utf8');
+  employees = data.split('\n').map((line) => {
+    const [name, email, age, dob] = line.split(',');
+    return new Employee(name, email, parseInt(age), dob);
+  });
+  return employees;}
+  catch(err){
+    if (err.code === 'ENOENT') {
+      console.log('File not found.');
+      return [];
+    }
+    else{
+    console.log("Error in reading file", err);
+    return [];}
+  }
+}
+    
+// Write employees to the file
+function writeEmployeesToFile(employees) {
+  try{
+  const data = employees
+    .map((employee) => {
+      const { name, email, age, dob } = employee;
+      
+      return `${name},${email},${age},${dob}`;
+    })
+    .join('\n');
+  fs.writeFileSync(fileName, data);}
+  catch(err){
+
+    console.log("Error in writing file", err);
+  }
+}
+
+// Validate the date of birth
+function isValidDOB(dob){
+
+  if (!/^\d{4}-\d{2}-\d{2}$/.test(dob)) {
+  console.log('\nDate of birth should be in the format YYYY-MM-DD.');
+  return false;
+}
+
+const [year, month, day] = dob.split('-').map(Number);
+
+if (isNaN(year) || isNaN(month) || isNaN(day)) {
+  console.log('\nDate of birth should contain only numbers.');
+  return false;
+}
+
+if (month < 1 || month > 12) {
+  console.log('\nMonth should be between 1 and 12.');
+  return false;
+}
+
+if (day < 1 || day > 31) {
+  console.log('\nDay should be between 1 and 31.');
+  return false;
+}
+
+return true;
+
+}
+      
+
+function displayEmployees() {
+
+  console.log('Employees:');
+  console.table(loadEmployeesFromFile());
+}
+
+
+  
+function menu(){
+// Define the main program
+loadEmployeesFromFile();
+
+rl.setPrompt('Menu:\n1. Add employee\n2. Delete employee\n3. Search employees\n4. Display \n5.Exit\nEnter your choice:');
+rl.prompt();
+
+rl.on('line', (choice) => {
+  switch (choice.trim()) {
+    case '1':
+      rl.question('Enter name:', (name) => {
+        rl.question('Enter email:', (email) => {
```

Comment:
Add this validation- if the entered email already exists because no two employees can have the same email.